picard/ui/metadatabox.py

```diff
@@ -268,12 +268,18 @@ def closeEditor(self, editor, hint):
     def contextMenuEvent(self, event):
         menu = QtGui.QMenu(self)
         if self.objects:
-            tags = self.selected_tags()
+            tags = self.selected_tags(discard=('~length'))
             if len(tags) == 1:
+                selected_tag = list(tags)[0]
                 edit_tag_action = QtGui.QAction(_(u"Edit..."), self.parent)
                 edit_tag_action.triggered.connect(partial(self.edit_tag, list(tags)[0]))
                 edit_tag_action.setShortcut(self.edit_tag_shortcut.key())
+                edit_tag_action.triggered.connect(partial(self.edit_tag, selected_tag))
                 menu.addAction(edit_tag_action)
+                if selected_tag in TAG_NAMES:
```

Comment:
Sorry, I think you added this check after my comment about enabling the action unconditionally. Looking back, that has to be one of the most vague comments I've ever written ;-) What I actually meant was that the action shouldn't be enabled or shouldn't be in the menu at all if the tag is already in the preserved tags list (the context menu here seems to use the latter option).